src/software/ai/intent/intent.h

```diff
@@ -3,8 +3,30 @@
 #include <set>
 #include <string>
 
-#include "software/ai/intent/intent_visitor.h"
+#include "shared/proto/primitive.pb.h"
 #include "software/ai/motion_constraint/motion_constraint.h"
+#include "software/geom/angle.h"
+#include "software/geom/point.h"
+#include "software/primitive/primitive.h"
+#include "software/proto/message_translation/proto_creator_primitive_visitor.h"
+
+MAKE_ENUM(BallCollisionType, AVOID, ALLOW);
```

Comment:
made https://github.com/UBC-Thunderbots/Software/issues/1651 as making this change is out of scope, since BallCollisionType is used in Actions when constructing Intents